src/test/java/io/confluent/examples/streams/IntegrationTestUtils.java

```diff
@@ -94,13 +96,13 @@
   public static <K, V> List<KeyValue<K, V>> readKeyValues(final String topic, final Properties consumerConfig, final int maxMessages) {
     final KafkaConsumer<K, V> consumer = new KafkaConsumer<>(consumerConfig);
     consumer.subscribe(Collections.singletonList(topic));
-    final int pollIntervalMs = 100;
-    final int maxTotalPollTimeMs = 2000;
+    final Duration pollInterval = Duration.ofMillis(100L);
+    final int maxTotalPollTimeMs = 10000;
```

Comment:
The new `Consumer.poll(Duration)` has different semantics that the deprecated `poll(long)` -- needed to increase the timeout to make tests passing locally.

Reply:
I think if we adjust the way we increment `totalPollTimeMs`, we can leave this maximum alone.

Reply:
Can you elaborate?

Reply:
That does not work. I dug into this a little bit, and the old `poll(long)` takes about 3000ms the first time it's called. Hence, the timeout of 2000ms is too low for the new methods that block about 100ms the first time. Getting the metadata from the brokers is just more overhead we need to factor in.

Reply:
I see. Thanks for the explanation!